src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -980,6 +965,32 @@ private void SkipWhiteSpace()
             }
         }
 
+        private void ScanNewline(char c)
```

Comment:
The name confuse me. We do something only if TokenList != null.

Reply:
When I started working on this code, there were a bunch of Scan* and Skip* methods, but the way those verbs were used was not consistent. I normalized this such that Scan* methods are used to scan characters and add new tokens, and Skip* methods are used to skip characters.

The check for `TokenList != null` was there before this PR, so I kept it, assuming it was necessary. Since you called it out, I'm now curious about what might cause the situation where a Scan* method is invoked when TokenList is null. I'll think about this and check it when I get back to my Windows laptop where I have that PR already to test/debug.

Reply:
Maybe @daxian-dbw have ideas how better name the methods.

Reply:
Yeah, since we're discussing this, it seems like a bug to me as well, because the method would be called and if TokenList was null, do nothing. Not advance the character marker, not add the token. That logic was in place before, so this is a legacy problem. I'm at a conference right now (PowerShell and DevOps Summit), so distracted, but I will revisit that code soon.

Reply:
Regarding TokenList, I kept the condition so that memory optimizations remain (it was there before), but I modified the logic so that only the token creation is skipped if TokenList is null (this eliminates what felt like a bug where CR in CRLF wouldn't be skipped if TokenList was null).

Reply:
Regarding the name of the method, I went with NormalizeCRLF, since that's exactly what the method does. SkipCR was not appropriate because it only skips CRs that are followed by LF.